tensorflow/contrib/data/python/ops/dataset_ops.py

```diff
@@ -428,6 +429,21 @@ def batch(self, batch_size):
     """
     return Dataset(dataset_ops.BatchDataset(self._dataset, batch_size))
 
+  def slide(self, slide_size, slide_step=1):
+    """Creates a sliding window on this dataset.
+
+    Args:
+      slide_size: A `tf.int64` scalar `tf.Tensor`, representing the number of
+        elements in the window.
+      slide_step: (Optional.) A `tf.int64` scalar `tf.Tensor`, representing
+        the steps moving forward for one iteration. The default is `1`. It
+        must be in [1, `slide_size`).
+
+    Returns:
+      A `Dataset`.
+    """
+    return Dataset(sliding.SlideDataset(self, slide_size, slide_step))
+
```

Comment:
Please revert this file. The `tf.contrib.data.Dataset` class is deprecated, and we don't want to encourage any new users :). Using `Dataset.apply()` together with a `tf.contrib.data` function defined in `sliding.py` should suffice.